CLDR-18144 Add to languagelocale-names.md for core/extension (#4601)
unicode-org/cldr: docs/site/translation/displaynames/languagelocale-names.md

If your standard translation of the language name already puts the family name first (as in “Kurdish, Central”) then you can supply the same name as the menu variant.

### Core and Extensions

Sometimes languages and other fields need further qualitification in the context of a menu. For example, when there is only one variant of French shown in a menu, it is fine to just list the language as follows.
* English
* French
* German

> However, when there are multiple variants, it is clearer to list each one, such as:

Reviewer: Minor copy-edit to clarify pronouns but you can leave as-is if you like.
```suggestion
However, when there are multiple variants, it is clearer to show the variant information, such as:
```

* English
* French (Canada)
* French (France)
* German

> When the variants differ not by region or script, but by another qualifier, then you may see codes with "-core" or "-extension", as in the following.

Reviewer: You should probably also indicate that variants may appear with different language codes. It's not a matter of variants for Kurdish since the language codes are "ku" and "ckb" -- rather the issue is menu item collision since the core for each is "Kurdish".

However, Kurdish may be a bad example since the languages are so different (and ku is in Latin script, ckb is in Arabic script) it may always be preferable to show Kurdish (Kurmanji) and Kurdish (Sorani).

Author: I do this in the example now.

Author: I also revised the other text; please see if that is clear enough.

<img src="kurdish-example.png" />

In these cases, supply the translated name of the qualifier (Kurmanji), and of the core.

> The core should omit the qualification (if any), and the extension should include it. For example, for Sorani Kurdish, you would see:

Reviewer: The algorithm in prose is a bit hard to follow. This makes me wonder why we have the base (plain `ckb` value) in the first place -- for versions that don't support algorithms?

1. Check if _lang_-core exists
   a. Use _lang_-core if unique
   b. Use _lang_-core (_lang_-extension) if it conflicts and extension clarifies
   c. Use _lang_-core (_region_) if it conflicts and the region clarifies
2. Otherwise, use plain _lang_
   a. Use _lang_ (_region_) if it conflicts and the region clarifies

Author: This isn't intended to be an algorithm for when to show what — that will be in the spec, and is too detailed for most vetters. There is an outline of that in the bug (but again, not appropriate for vetters).

The text is just to (1) set the expectation for the vetter as to what the appearance is if the regular value is used vs what it is when the core + extension is used, and (2) give an example of when the core + extension could appear.

I'll play with the wording a bit.
